xmtp_mls/src/groups/decoded_message.rs

```diff
@@ -86,9 +79,123 @@ pub struct DecodedMessage {
     // The content of the message
     pub content: MessageBody,
     // Fallback text for the message
-    pub fallback_text: String,
+    pub fallback_text: Option<String>,
     // A list of reactions
-    pub reactions: Vec<Reaction>,
+    pub reactions: Vec<DecodedMessage>,
     // The number of replies to the message available
     pub num_replies: usize,
 }
+
+impl TryFrom<EncodedContent> for MessageBody {
+    type Error = GroupError;
+
+    fn try_from(value: EncodedContent) -> Result<Self, Self::Error> {
+        let content_type = match value.r#type.as_ref() {
+            Some(content_type) => content_type,
+            None => return Err(CodecError::InvalidContentType.into()),
+        };
+
+        match (content_type.type_id.as_str(), content_type.version_major) {
```

Comment:
If the constants where on a trait, we could avoid this match statement altogether, and use the generic constants, as long as we require that a content type implement the prost::Message type